PERFORMANCE.md

```diff
@@ -0,0 +1,127 @@
+
+#### Put Tensor - send image to database
+![Throughput colo put](/figures/put_tensor_thro_colo.png "Colocated Throughput")
+
+## Throughput Performance Analysis
+
+1. The Throughput Standard test density decreases as the database nodes increase. Outliers become greater as client total increases. This is because as there are more clients, there is a longer wait time. We can also tell that the standard takes longer than colocated because the client nodes are having to travel offsight to the database client node. Each database is having to handle more requests since they do not 
+have their own clients. 
+
+2. Put/Unpack tensor for Throughput Colocated stays consistent with density across all client totals.
+It takes much less time to complete using a colocated orchestrator since all the computations are happening on the same node. Less time it takes to communicate since the db and app are on the same node.
+
+## Data Aggregation Standard
+
+The following are scaling results for a data aggregation test, run with 32 client threads on each node. The databases were run on 36 CPUs and 4 threads per dataset. 
+
+#### Get List - retrieve the data from the aggregation list
+![Data Agg Get List](/figures/get_list_data_agg.png "Data Aggregation Standard")
+
+## Data Aggregation Standard Py
+The following are scaling results for a data aggregation py test, run with 32 client threads on each node. The databases were run on 36 CPUs and 4 threads per dataset.
+
+#### Get List - retrieve the data from the aggregation list
+![Data Agg Py Get List](/figures/get_list_data_agg_py.png "Data Aggregation Py Standard")
+
+#### Poll List - check when the next aggregation list is ready
+![Data Agg Py Poll List](/figures/poll_list_data_agg_py.png "Data Aggregation Py Standard")
+
+## Data Aggregation Standard Py Fs
+The following are scaling results for a data aggregation py fs test, run with 32 client threads on each node.
+
+#### Get List - retrieve the data from the aggregation list
+![Data Agg Py Fs Get List](/figures/get_list_data_agg_fs.png "Data Aggregation Py Fs Standard")
+
+#### Poll List - check when the next aggregation list is ready
+![Data Agg Py Fs Poll List](/figures/poll_list_data_agg_fs.png "Data Aggregation Py Fs Standard")
+
+## Data Aggregation Performance Analysis
+
+1. The Standard Data Aggregation test performs the fastest get_list() client function. We
+can assume this is due to the fact that there is one less communication layer
+than the other two tests that use python consumer clients with c++ producer clients. 
+We can also see that as the database node count increases, the violin plot
+density decreases. We can assume this is happening because there are more shards
+of the database avaiable to complete the function get_list().
+
+2. The Standard Data Aggregation Python test seems to be less stable than the previous test using all C++ clients. Elements of the list might be distributed unevenly causing a contention between this test and the
```

Comment:
I don't think I agree with this argument. The Python client and the C++ argument are algorithmically equivalent, so it doesn't make that much sense that one would preferentially be distributed differently than the other. 

Also, the phrase "contention between this test and the previous" makes it seem that these two tests are _not_ independent of each other. They should be completely independent tests

Reply:
okay I agree with this, Im trying to write a less kindergarten summary lol

Reply:
actually lowkey the cpp get_list with a python client looks a lil more wonky, like it took a lil more time, also the y axis is .08 seconds higher than the cpp c++ client